### DISCUSSION TITLE
PointSet and OrthTree

### DISCUSSION THREAD

**Discussion by petrasvestartas** [CGAL/cgal#6475] [Q&A]
Hi,

Is there any example how you can apply 3d spatial tree for pointset distance search?

For example I contruct a pointset like this: https://doc.cgal.org/latest/Point_set_3/Point_set_3_2draw_point_set_3_8cpp-example.html

But for Orthtree I need to construct tree from scratch. 
https://doc.cgal.org/latest/Orthtree/index.html

So should I use pointsets or I must construct trees instead? Are there any spatial searching algorithms in CGAL applicable to pointsets directly?

**Comment by afabri** [comment]
I do not really understand what you want to achieve.   Did you have a look at the [Spatial Searching](https://doc.cgal.org/latest/Manual/packages.html#PkgSpatialSearchingD) package?

**Reply by petrasvestartas** [reply]
Dear @afabri ,

What I am trying to do is to crop one point cloud from the other, like the image below. It is the same problem as before:
1) SetA and SetB are on the same location. 
2) I need to search all points from one pointcloud to the other.
3) Then remove points from a pointcloud given a certain distance.

I will try to use Spatial Searching by testing the example files.
I thought that point_set data structures have methods for spatial distance searches.
![image](https://user-images.githubusercontent.com/18013985/162022394-c354d102-89f7-4687-91a4-ee581bd01f4d.png)

**Reply by petrasvestartas** [reply]
Dear @afabri ,

In this example closest points are returned. 
https://doc.cgal.org/latest/Spatial_searching/Spatial_searching_2parallel_kdtree_8cpp-example.html
Is there an example that outputs point indices if they point are closer than a given distance?

**Comment by afabri** [comment]
You want maybe do a range query that is retreiving all points at a distance less than r   (which is a sphere of radius r).

**Reply by petrasvestartas** [reply]
Dear @afabri ,

Yes, exactly.
Could you point me to a correct documentation page for the radius search?
I am a bit lost in a number of different examples.

**Comment by afabri** [comment]
Have a look at this  [example ](https://doc.cgal.org/latest/Spatial_searching/Spatial_searching_2circular_query_8cpp-example.html) in 2D that you have to adapt to 3D.
